### PR TITLE
gh-94808: add tests covering `PyFunction_{Get,Set}Closure`

### PR DESCRIPTION
**Pull request by sobolevn**
It was like this:
Снимок экрана 2022-11-13 в 13 00 40

Now all branches are covered.

* Issue: gh-94808

### CODE REVIEW

File: Modules/_testcapimodule.c [hunk 1]
```diff
@@ -6009,6 +6009,33 @@ function_set_kw_defaults(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+function_get_closure(PyObject *self, PyObject *func)
+{
+    PyObject *closure = PyFunction_GetClosure(func);
+    if (closure != NULL) {
+        Py_INCREF(closure);
+        return closure;
```
Comment (kumaraditya303) [inline Modules/_testcapimodule.c]:
```suggestion
        return Py_NewRef(closure);
```

**Review by kumaraditya303 — DISMISSED** [review]
This has merge conflicts now.

**Comment by encukou** [comment]
After a year, I took the liberty of updating to the current main branch, and fixed two nitpicks.

**Review by gvanrossum — APPROVED** [review]

**Comment by sobolevn** [comment]
@encukou thanks a lot for your help! I forgot about this PR a long time ago :)

**Pull request merged**